pkg/generate/cspc.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2020-2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generate
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	cstorv1 "github.com/openebs/api/v2/pkg/apis/cstor/v1"
+	"github.com/openebs/api/v2/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/openebsctl/pkg/client"
+	"github.com/openebs/openebsctl/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// isPoolValid checks if a CStor pool is valid
+func isPoolTypeValid(raid string) bool {
+	if raid == "stripe" || raid == "mirror" || raid == "raidz" || raid == "raidz2" {
+		return true
+	} else {
+		return false
+	}
+}
+
+// Pool calls the generate routine for different cas-types
+func Pool(nodes []string, devs int, raid string) error {
```

Comment:
nit: Good to have function name related to CSPC/cStor pool
```suggestion
func CStorPool(nodes []string, devs int, raid string) error {
```

Reply:
I have this function to call different generator functions for `cstor`, but it seems pool concept as such isn't there on non-cstor nodes wherein writing Pool YAMLs is complex.

Reply:
I've renamed the functions to `CSPC` and `cspc`, they aren't good names, but having a function create a K8s-client object helps me better unit test it.

Background: The CSPC function is a dumb one, it does two things
1. Create a client object
2. Call the `cspc` which does interesting stuff and returns the `err` which is printed to the console.

Benefits: I can easily write the tests for business logic( which is in the `cspc(....)` and `CSPC()` has stuff which don't require tests). This way the `cmd` package has no k8s dependency and `cmd` package tests can also be written as an overkill someday, moreover, someone can import `openebsctl` as a dependency and call some of it's function(won't happen anyday I suppose).